OPHD/States/MapViewStateHelper.cpp

```diff
@@ -516,6 +516,10 @@ void checkRobotDeployment(XmlElement* _ti, RobotTileTable& _rm, Robot* _r, Robot
 	_ti->attribute("age", _r->fuelCellAge());
 	_ti->attribute("production", _r->turnsToCompleteTask());
 
+	if (_rm.empty()) {
+		return;
+	}
+
 	const auto it = _rm.find(_r);
 	if (it->first == _r)
```

Comment:
We should probably check validity of `it` after the call to `find`. It's possible we could be looking up an item that doesn't not exist in the collection, even though the collection is non-empty.

According to [`std::map::find`](https://en.cppreference.com/w/cpp/container/map/find), we should compare the iterator against `_rm.end()` before  using it. If it's not equal to `.end()`, it will be valid, and the key will match `_r`. That would also make the check more consistent with other similar checks spread throughout the code.

```cpp
const auto it = _rm.find(_r);
if (it != _rm.end())
{
    // ...
```

With an improved check on the result of `find`, we shouldn't need to check if the collection was empty or not.